crates/bevy_math/src/common_traits.rs

```diff
@@ -161,3 +161,142 @@ impl NormedVectorSpace for f32 {
         self * self
     }
 }
+
+/// A type with a natural interpolation that provides strong subdivision guarantees.
+///
+/// Although the only required method is `interpolate_stable`, many things are expected of it:
+///
+/// 1. The notion of interpolation should follow naturally from the semantics of the type, so
+///    that inferring the interpolation mode from the type alone is sensible.
+///
+/// 2. The interpolation recovers something equivalent to the starting value at `t = 0.0`
```

Comment:
I chose this loosey-goosey wording around "equivalence" because what I really mean is that they don't necessarily have to be data-identical, but they do have to be semantically identical. An example of this is that a `Quat` as used in `glam`/`bevy_math` represents a rotation, and it's the case that `slerp` doesn't always return a data-identical quaternion at its end; however, it does always return one that represents the same rotation as the one that was input. I'll amend the docs to make this more clear and make the two more consistent.

Reply:
Good point on `Quat` rotations. Maybe it's worth writing down that example to explain the wording.

Reply:
I like the new clarification you added here.